src/software/new_geom/polygon.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <vector>
+
+#include "software/new_geom/segment.h"
+#include "software/new_geom/shape.h"
+
+/**
+ * A shape composed of segments.
+ */
+class Polygon : public virtual Shape
+{
+   public:
+    Polygon() = delete;
+    /**
+     * Construct a polygon by drawing line segments between consecutive
+     * Points, and then from the last Point to the first Point.
+     * @param points Points that form a polygon
+     */
+    explicit Polygon(const std::vector<Point>& points);
+
+    /**
+     * Construct a polygon by drawing line segments between consecutive
+     * Points, and then from the last Point to the first Point.
+     * @param points Points that form a polygon
+     */
+    explicit Polygon(const std::initializer_list<Point>& points);
+
+    /**
+     * Determines whether the given Point is contained within this Polygon.
+     *
+     * @return whether the Point p is contained within this Polygon.
+     */
+    bool contains(const Point& p) const override;
+
+    /**
+     * Returns the line segments that form this polygon.
+     * @return the line segments that form this polygon.
+     */
+    const std::vector<Segment>& getSegments() const;
+
+    /**
+     * Returns the points that form the polygon.
+     * @return the points that form the polygon.
+     */
+    const std::vector<Point>& getPoints() const;
+
+   protected:
+    std::vector<Segment> segments_;
```

Comment:
Agreed, I also think points are a better representation because it's not possible to construct a non-closed shape with them. Totally fine to have a `getSegments` method still.

Reply:
Addressed in latest update